src/content/tutorials/user-auth-with-auth0/_index.md

```diff
@@ -0,0 +1,736 @@
+
+  const text = new TextEncoder().encode(`${SALT}-${decoded.sub}`)
+  const digest = await crypto.subtle.digest({ name: 'SHA-256' }, text)
+  const digestArray = new Uint8Array(digest)
+  const id = btoa(String.fromCharCode.apply(null, digestArray))
+
+  await AUTH_STORE.put(id, JSON.stringify(body))
+}
+```
+
+Once the user's authentication data has been stored in KV, we need to associate the user with that data. To do this, we'll set a cookie, setting the value to the encrypted `id` string we just defined.
+
+This cookie will be used as we fill out the `verify` function defined earlier in the tutorial, and we'll set it to expire in a day (though this is easily customizable to your application's needs). To persist this cookie, we'll return an object containing some data for a [Response](https://developer.mozilla.org/en-US/docs/Web/API/Response) instance, redirecting the user via an [HTTP 302 response](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/302) to the `/` path, with a `Set-cookie` header:
+
+```js
+// workers-site/auth0.js
+
+const cookieKey = 'AUTH0-AUTH'
+
+const persistAuth = async exchange => {
+  // previous code
+
+  const date = new Date()
+  date.setDate(date.getDate() + 1)
+
+  const headers = {
+    Location: '/',
+    'Set-cookie': `${cookieKey}=${id}; HttpOnly; SameSite=Lax; Expires=${date.toUTCString()}`,
+  }
+
+  return { headers, status: 302 }
+}
+```
+
+With our authorization logic defined, we can finally wrap up the corresponding code in `workers-site/index.js`. Knowing that the `handleRedirect` function will pass back an object with `Response` options, we can make a new `Response` object, passing in the information from the existing `response`, and adding the new headers and status code from `handleRedirect`. This will redirect the user to the app's root path, setting a cookie to indicate that they're authorized for future requests:
+
+```js
+// workers-site/index.js
+
+async function handleEvent(event) {
+  try {
+    // BEGINNING OF HANDLE AUTH REDIRECT CODE BLOCK
+    if (url.pathname === '/auth') {
+      const authorizedResponse = await handleRedirect(event)
+      response = new Response(response.body, {
+        response,
+        ...authorizedResponse,
+      })
+      return response
+    }
+    // END OF HANDLE AUTH REDIRECT CODE BLOCK
+
+    // BEGINNING OF WORKERS SITES
+  }
+}
+```
+
+## Adding the verify read block
+
+With our application persisting authentication data in KV, and associating it to the current user via a cookie, we're now prepared to fill out the `verify` function defined earlier in the tutorial. This function will look at the `Cookie` header, and look up the authentication info that we persisted in Workers KV. The tokens saved in Workers KV can be used to make a subrequest to Auth0, which has the benefit of both re-verifying the user, as well as getting some user info, such as name and email address, if they're provided using the user's authentication method of choice.
```

Comment:
Theres a lot going on conceptually in this last sentence. Could it be simplified a little?